Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -57,54 +57,78 @@ public class ResourcesProjectMapper: ProjectMapping {
             .appending(component: "Bundle+\(target.name).swift")
 
         let content: String = ResourcesProjectMapper.fileContent(targetName: target.name,
-                                                                 bundleName: bundleName)
+                                                                 bundleName: bundleName,
+                                                                 target: target)
         return (filePath, [.file(.init(path: filePath, contents: content.data(using: .utf8), state: .present))])
     }
 
-    static func fileContent(targetName: String, bundleName: String) -> String {
-        """
-        // swiftlint:disable all
-        import Foundation
-
-        // MARK: - Swift Bundle Accessor
-
-        private class BundleFinder {}
-
-        extension Foundation.Bundle {
-            /// Returns the resource bundle associated with the current Swift module.
-            static var module: Bundle = {
-                let bundleName = "\(bundleName)"
-
-                let candidates = [
-                    // Bundle should be present here when the package is linked into an App.
-                    Bundle.main.resourceURL,
-
-                    // Bundle should be present here when the package is linked into a framework.
-                    Bundle(for: BundleFinder.self).resourceURL,
-
-                    // For command-line tools.
-                    Bundle.main.bundleURL,
-                ]
-
-                for candidate in candidates {
-                    let bundlePath = candidate?.appendingPathComponent(bundleName + ".bundle")
-                    if let bundle = bundlePath.flatMap(Bundle.init(url:)) {
-                        return bundle
+    static func fileContent(targetName: String, bundleName: String, target: Target) -> String {
+        if !target.supportsResources {
+            return """
+            // swiftlint:disable all
+            import Foundation
+
+            // MARK: - Swift Bundle Accessor
+
+            private class BundleFinder {}
+
+            extension Foundation.Bundle {
+                /// Since \(targetName) is a library, the bundle containing the resources is copied into the final product (app).
```

Comment:
minor:

similarly here: the final product could be a framework - I guess we can omit the `(app)`?

> Since `\(targetName)` is a `\(target.product)`, the bundle containing the resources is copied into the final product.